templates/funcs.go

```diff
@@ -172,11 +172,20 @@ func newArrayData(sliceType string, getter string, key string) ArrayData {
 }
 
 const arrayTpl = `
-{{ .SliceName }} := make(utils.{{ .SliceType }}, len({{ .Getter }}))
-for {{ .IndexName }}, {{ .ItemName }} := range {{ .Getter }} {
-	{{ .SliceName }}[{{ .IndexName }}] = {{ .ItemName }}
+
+{{ .Key }}Length := len({{ .Getter }})
+o.AddInt("{{ .Key }}_length", {{ .Key }}Length)
+
+if {{ .Key }}Length > 100 {
+	{{ .Key }}Length = 100
+}
+
+{{ .SliceName }} := make(utils.{{ .SliceType }}, {{ .Key }}Length)
+for i := 0; i < {{ .Key }}Length; i++ {
+	{{ .SliceName }}[i] = {{ .Getter }}[i]
 }
```

Comment:
Using .Key makes your code not very go-like, because if my key is `line_items`, you will call the variable `line_itemsLength`